### PR TITLE
Fix NFT visibility in BottomSheet fragment

### PR DESCRIPTION
**Pull request by simoarpe**
## Details

Fixes an issue where NFTs listed in `EditVisibleAssetsBottomSheetDialogFragment` were always marked as visible.

Screen recording showing the right behavior:

https://github.com/brave/brave-core/assets/3308503/b5811252-b1e0-405d-8c0e-ac4eaa94106a

## ⚠️ Notes for Reviewers

The approach used to mark an asset as visible or invisible is going to be rewritten soon. 
The new way will leverage a WebUI based on Desktop implementation and will (probably) rely on the same persistency methods if compatible with Android. 
Because we don't want to reinvent the wheel, the actual fix is just a simple check on the NFT data model returned from core API and is implemented by the commit 👉 https://github.com/brave/brave-core/pull/20361/commits/357eaf4e0ecddd51dcbc519403f4be4db66aa01e
After the transition to a WebUI the state will be handled on the client side and not from core.

## Other Notable Changes

* `NetworkUtils#findNetwork` has been improved with a new parameter as localhost networks now can belong to FIL, SOL or ETH and on debug builds was returning wrong values.
* Improved `filterTokens` method: the filtering now uses Java streams, and it's (roughly) 50% faster as the previous logic was converting to/from array/list multiple times.
* Added Javadoc to `TokenUtils` methods.

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

* Open Brave Wallet and navigate to NFT section by tapping on the selector to the right.
* If not present, add some NFTs by tapping on the plus icon.
* Tap on "Edit visible NFTs" and deselect all NFTs.
* Observe the full list is shown in the Hidden section.
* Tap on "Edit visible NFTs" and deselect all NFTs.
* Select all the available NFTs.
* Observe the full list is shown in the NFTs section.
* Observe the Hidden section is empty.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/app/domain/UserAssetModel.java [hunk 1]
```diff
@@ -63,34 +62,20 @@ public void fetchAssets(boolean nftsOnly, NetworkInfo selectedNetwork) {
...
-                                    mSelectedNetwork.coin, TokenUtils.TokenType.ALL, tokens -> {
-                                        _mAssetsResult.postValue(new AssetsResult(
-                                                Arrays.asList(tokens), Collections.emptyList()));
-                                    });
-                        } else if (mType == WalletCoinAdapter.AdapterType.BUY_ASSETS_LIST) {
-                            TokenUtils.getBuyTokensFiltered(mBlockchainRegistry, mSelectedNetwork,
-                                    TokenUtils.TokenType.ALL, tokens -> {
-                                        _mAssetsResult.postValue(new AssetsResult(
-                                                Arrays.asList(tokens), Collections.emptyList()));
-                                    });
```
Comment (simoarpe) [inline android/java/org/chromium/chrome/browser/app/domain/UserAssetModel.java:93]: `SEND_ASSETS_LIST` and `BUY_ASSETS_LIST` can be safely removed as they were never accessed: these parts now rely on WebUIs.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by Pavneet-Sing** [comment]
> The approach used to mark an asset as visible or invisible is going to be rewritten soon.
The new way will leverage a WebUI based on Desktop implementation and will (probably) rely on the same persistency methods if compatible with Android.
Because we don't want to reinvent the wheel, the actual fix is just a simple check on the NFT data model returned from core API and is implemented by the commit 👉 https://github.com/brave/brave-core/commit/357eaf4e0ecddd51dcbc519403f4be4db66aa01e
After the transition to a WebUI the state will be handled on the client side and not from core.

We don't need to maintain anything or reinvent(as it is a bug). Token sates is definitely handled on the core side, not on client side. This seems like a regression for assets/tokens, but NFT visibility is completely depends on `isVisible` flag so this is the accurate and only required check to determine NFT visibility.

About tokens visibility issue, It is working as expected for non-native tokens. The issue with native token visibility is that the native token always seems to be added by [Stream.concat(Stream.of(nativeAsset), tokenStream);](https://github.com/brave/brave-core/blob/simone/fix-assets-visibility/android/java/org/chromium/chrome/browser/crypto_wallet/util/TokenUtils.java#L106) and that's why it is always visible.

[assets.webm](https://github.com/brave/brave-core/assets/11755381/a24762f2-1f39-4d3f-948b-1e322500f455)

Anyway, We will come back to these native changes/issues later if needed after completing webUI issues as priority.

**Review by Pavneet-Sing — APPROVED** [review]

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 13 running 1.61.3 x64 Nightly build
- Verified steps from PR
- Verified unchecking NFTs moves them into hidden tab
- Verified checking NFTs moves them back into collected tab

https://github.com/brave/brave-core/assets/17010094/01061bee-cc13-4703-b70e-53271e10e0a3

**Pull request merged**